src/main/java/seedu/address/model/CommandHistoryManager.java

```diff
@@ -21,6 +22,9 @@ public CommandHistoryManager() {
     public void saveCommandExecutionString(String commandInputString) {
         //Every execution of a new command will cause the currIndex to be reset
         //to the right-most index of the `history` + 1
+        if (this.history.size() == this.capacity) {
```

Comment:
btw since capacity is a static variable, you dont need to add a "this" in "this.capacity"